lib/phonebook/providers/association_list_provider.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:myecl/phonebook/class/association.dart';
+import 'package:myecl/phonebook/repositories/association_repository.dart';
+import 'package:myecl/auth/providers/openid_provider.dart';
+import 'package:myecl/tools/providers/list_notifier.dart';
+import 'package:myecl/tools/token_expire_wrapper.dart';
+
+class AssociationListNotifier extends ListNotifier<Association> {
+  final AssociationRepository associationRepository = AssociationRepository();
+  AsyncValue<List<Association>> associationList = const AsyncValue.loading();
+  AssociationListNotifier({
+    required String token,
+  }) : super(const AsyncValue.loading()) {
+    associationRepository.setToken(token);
+  }
+
+  Future<AsyncValue<List<Association>>> loadAssociations() async {
+    associationList = await loadList(() async => associationRepository.getAssociationList());
+    return associationList;
```

Comment:
This can also be shorten
return await loadList(associationRepository.getAssociationList);

Reply:
Fixed in next commit